hplot/binerrband.go

```diff
@@ -0,0 +1,109 @@
+// Copyright ©2020 The go-hep Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package hplot
+
+import (
+	"image/color"
+	"math"
+
+	"go-hep.org/x/hep/hbook"
+
+	"gonum.org/v1/plot"
+	"gonum.org/v1/plot/plotter"
+	"gonum.org/v1/plot/vg/draw"
+)
+
+// BinnedErrBand implements the plot.Plotter interface,
+// drawing a colored band for the error on any binned
+// quantity.
+type BinnedErrBand struct {
+
+	// Data for every bins.
+	Counts []hbook.Count
+
+	// LineStyle is the style of the line
+	// contouring the band.
+	// Use zero width to disable.
+	draw.LineStyle
+
+	// FillColor is the color to fill the area
+	// between the top and bottom data points.
+	// Use nil to disable the filling.
+	FillColor color.Color
+}
+
+// NewBinnedErrBand creates a binned error band
+// from a slice of count.
+func NewBinnedErrBand(h *hbook.H1D) *BinnedErrBand {
```

Comment:
I think it would probably make more sense to have that function take directly a slice of `hbook.Count`.